pvlib/iotools/sodapro.py

```diff
@@ -0,0 +1,368 @@
+
+    Notes
+    -----
+    In order to use the CAMS services, users must register for a free SoDa
+    account using an email address [1]_.
+
+    The returned data DataFrame includes the following fields:
+
+    ========================  ======  =========================================
+    Key, mapped key           Format  Description
+    ========================  ======  =========================================
+    **Mapped field names are returned when the map_variables argument is True**
+    ---------------------------------------------------------------------------
+    Observation period        str     Beginning/end of time period
+    TOA, ghi_extra            float   Horizontal radiation at top of atmosphere
+    Clear sky GHI, ghi_clear  float   Clear sky global radiation on horizontal
+    Clear sky BHI, bhi_clear  float   Clear sky beam radiation on horizontal
+    Clear sky DHI, dhi_clear  float   Clear sky diffuse radiation on horizontal
+    Clear sky BNI, dni_clear  float   Clear sky beam radiation normal to sun
+    GHI, ghi†                 float   Global horizontal radiation
+    BHI, bhi†                 float   Beam (direct) radiation on horizontal
+    DHI, dhi†                 float   Diffuse horizontal radiation
+    BNI, dni†                 float   Beam (direct) radiation normal to the sun
+    Reliability†              float   Reliable data fraction in summarization
+    ========================  ======  =========================================
+
+    †Parameters only returned if identifier='cams_radiation'. For description
+    of additional output parameters in verbose mode, see [1]_ and [2]_.
+
+    Note that it is recommended to specify the latitude and longitude to at
+    least the fourth decimal place.
+
+    Variables corresponding to standard pvlib variables are renamed,
+    e.g. `sza` becomes `solar_zenith`. See the
+    `pvlib.iotools.cams.CAMS_VARIABLE_MAP` dict for the complete
+    mapping.
+
+    See Also
+    --------
+    pvlib.iotools.read_cams, pvlib.iotools.parse_cams
+
+    Raises
+    ------
+    requests.HTTPError
+        If the request is invalid, then an XML file is returned by the CAMS
+        service and the error message will be raised as an exception.
+
+    References
+    ----------
+    .. [1] `CAMS Radiation Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-radiation-service/info>`_
+    .. [2] `CAMS McClear Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-mcclear/info>`_
+    .. [3] `CAMS McClear Automatic Access
+       <http://www.soda-pro.com/help/cams-services/cams-mcclear-service/automatic-access>`_
+    """
+    try:
+        time_step_str = TIME_STEPS_MAP[time_step]
+    except KeyError:
+        raise ValueError(f'Time step not recognized. Must be one of '
+                         f'{list(TIME_STEPS_MAP.keys())}')
+
+    if (verbose is True) & ((time_step != '1min') | (time_ref != 'UT')):
```

Comment:
Style nit:  I reserve the bitwise operators `&` and `|` for situations where I actually need the difference between the standard logical `and` and `or`, e.g. combining boolean vectors, so that it is marginally easier to tell when something is vectorized vs when it's just a scalar comparison.  Also minor preference to just use `verbose` rather than `verbose is True`, unless it's important here that the value is actually `True` instead of just something truthy?

Reply:
Consider it implemented!